File: Client/src/app/models/journal.ts
import { User } from "./user"
import { Trade } from "./trade";

export interface Journal {
    id: number;
    name: string;
    description: string;
    dateCreated: Date;
    appUser: User | null;
    appUserId: number;
    trades?: Trade[] | null;
}

export interface JournalDto {
    name: string;
    description: string;
}

File: Api/Models/Journal.cs
namespace Api.Models;

public class Journal
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime DateCreated { get; set; } = DateTime.Now;
    public AppUser AppUser { get; set; }
    public string AppUserId { get; set; }
    public ICollection<Trade> Trades { get; set; }
}
